### DISCUSSION TITLE
Workflow is running twice when creating a new branch

### DISCUSSION THREAD

**Discussion by ingomohr** [community/community#25996] [Actions]
<p>I have a workflow that is supposed to run whenever</p>
<ul>
<li>I create a new branch OR</li>
<li>I push some commits to a branch that is not the “master”-branch.</li>
</ul>
<p>What I’ve come up so far is this:</p>
<pre><code class="lang-auto">on:
  push:
    branches-ignore:
      - master
    tags-ignore:
      - '**'
  create:
    type:
      - branch
</code></pre>
<p>It works for both cases I’ve listed above, BUT…<br>
Whenever I create a new branch (without any other change to it), the workflow is triggered twice (at the same time):</p>
<ul>
<li>one for event “create”</li>
<li>one for event “push”</li>
</ul>
<p>Question:<br>
How can I make it run only once when I create a new branch?</p>
<p>Thanks in advance for any help!</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Creating a new branch usually involves creating it locally and then pushing it, that’s why the <code>push</code> event triggers in addition to <code>create</code>. So the solution is: Drop the <code>create</code> event from the workflow triggers. :slightly_smiling_face:</p>
<p>The <code>create</code> event is useful if you want a workflow that runs <em>only</em> for new branches.</p>

**Comment by ingomohr** [comment]
<p><a class="mention" href="/airtower-luna">@airtower-luna</a> Thank you! Works like a charm :slight_smile:</p>
